tst/EnergyPlus/unit/HVACMultiSpeedHeatPump.unit.cc

```diff
@@ -1502,4 +1503,734 @@ TEST_F(EnergyPlusFixture, HVACMultiSpeedHeatPump_HeatRecoveryTest)
+    msHeatPump.MyEnvrnFlag = true;
+    // verify min OAT from coil inputs
+    EXPECT_EQ(msHeatPump.MinOATCompressorCooling, -25.0);
+    EXPECT_EQ(msHeatPump.MinOATCompressorHeating, -8.0);
+    // set local variables for convenience
+    auto &supplyFan = state->dataFans->Fan(1);
+    auto &dxClgCoilMain = state->dataDXCoils->DXCoil(1);
+    auto &dxHtgCoilMain = state->dataDXCoils->DXCoil(2);
+    auto &elecHtgCoilSupp = state->dataHeatingCoils->HeatingCoil(msHeatPump.SuppHeatCoilNum);
+    state->dataScheduleMgr->Schedule(11).CurrentValue = 1.0;
+    state->dataEnvrn->StdRhoAir = 1.2;
+    supplyFan.RhoAirStdInit = state->dataEnvrn->StdRhoAir;
+    state->dataGlobal->DoCoilDirectSolutions = false;
+    state->dataGlobal->BeginEnvrnFlag = true;
+    state->dataGlobal->DoCoilDirectSolutions = false;
+    // set outdoor air conditions
+    state->dataEnvrn->OutDryBulbTemp = 5.0;
+    state->dataEnvrn->OutHumRat = 0.005;
+    state->dataEnvrn->StdBaroPress = 101325.0;
+    state->dataEnvrn->OutBaroPress = 101325.0;
+    // set zone air conditions
+    auto &zoneAirNode =
+        state->dataLoopNodes->Node(UtilityRoutines::FindItemInList("Z401 AIR NODE", state->dataLoopNodes->NodeID, state->dataLoopNodes->NumOfNodes));
+    zoneAirNode.Temp = 21.1;
+    zoneAirNode.HumRat = 0.0035;
+    zoneAirNode.Enthalpy = Psychrometrics::PsyHFnTdbW(zoneAirNode.Temp, zoneAirNode.HumRat);
+    // set maixed air node conditions
+    auto &mixedAirNode = state->dataLoopNodes->Node(
+        UtilityRoutines::FindItemInList("AC-24 SF INLET AIR NODE", state->dataLoopNodes->NodeID, state->dataLoopNodes->NumOfNodes));
+    mixedAirNode.Temp = 10.0;
+    mixedAirNode.HumRat = 0.005;
+    mixedAirNode.Enthalpy = Psychrometrics::PsyHFnTdbW(mixedAirNode.Temp, mixedAirNode.HumRat);
+    // init ms heat pump
+    HVACMultiSpeedHeatPump::InitMSHeatPump(*state, MSHeatPumpNum, FirstHVACIteration, AirLoopNum, QZnReq, OnOffAirFlowRatio);
+    // run the unit at zero load
+    Real64 QSensUnitOut = 0.0;
+    SimMSHP(*state, MSHeatPumpNum, FirstHVACIteration, AirLoopNum, QSensUnitOut, 0.0, OnOffAirFlowRatio);
+    // set zone heating load
+    QZnReq = zoneSysEnergyDemand.RemainingOutputRequired;
+    msHeatPump.HeatCoolMode = EnergyPlus::HVACMultiSpeedHeatPump::ModeOfOperation::HeatingMode;
+    SimMSHP(*state, MSHeatPumpNum, FirstHVACIteration, AirLoopNum, QSensUnitOut, QZnReq, OnOffAirFlowRatio);
+    // calculate the total electricity rate
+    Real64 result_msHeatPump_ElectricityRate = supplyFan.FanPower + state->dataHVACGlobal->DXElecCoolingPower +
+                                               state->dataHVACGlobal->DXElecHeatingPower + state->dataHVACGlobal->ElecHeatingCoilPower +
+                                               state->dataHVACGlobal->SuppHeatingCoilPower + msHeatPump.AuxElecPower;
+    // test results
+    EXPECT_NEAR(55366.46, msHeatPump.ElecPower, 0.01);
+    EXPECT_NEAR(55366.46, result_msHeatPump_ElectricityRate, 0.01);
```

Comment:
This looks to be the actual test of electricity rate.  I personally wouldn't mind the tolerances being loosened slightly.  The simulation accuracy is not meaningful down to the centiwatt, but this is fine for now.  Definitely an item for another day.